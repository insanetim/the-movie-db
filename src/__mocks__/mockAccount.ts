import { IAccount } from 'src/interfaces/account.interface'

const mockAccount: IAccount = {
  avatar: {
    gravatar: {
      hash: '/image'
    },
    tmdb: {
      avatar_path: null
    }
  },
  id: 1234,
  include_adult: true,
  iso_639_1: 'en',
  iso_3166_1: 'en',
  name: '',
  username: 'test/username'
}

export default mockAccount
