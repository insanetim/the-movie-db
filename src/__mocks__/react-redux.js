export const dispatch = jest.fn()

export const useDispatch = jest.fn(() => dispatch)
export const useSelector = jest.fn(fn => fn())
