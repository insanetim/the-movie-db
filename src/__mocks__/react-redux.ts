const dispatch = jest.fn()
const getState = jest.fn()

const useDispatch = jest.fn(() => dispatch)
const useSelector = jest.fn(fn => fn())

export { dispatch, getState, useDispatch, useSelector }
