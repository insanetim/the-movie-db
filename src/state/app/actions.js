import { v4 as uuid } from 'uuid'

import { NOTIFICATION_DURATION, NOTIFICATION_TYPE } from 'src/constants'

import * as types from './types'

export const showModal = payload => ({
  type: types.SHOW_MODAL,
  payload
})

export const hideModal = () => ({
  type: types.HIDE_MODAL
})

export const showNotification = ({
  id = uuid(),
  messageType = NOTIFICATION_TYPE.SUCCESS,
  messageText,
  duration = NOTIFICATION_DURATION
}) => ({
  type: types.SHOW_NOTIFICATION,
  payload: {
    id,
    messageType,
    messageText,
    duration
  }
})

export const hideNotification = payload => ({
  type: types.HIDE_NOTIFICATION,
  payload
})
