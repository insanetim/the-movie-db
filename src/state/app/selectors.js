import { path } from 'ramda'

export const modalTypeSelector = path(['app', 'modal', 'modalType'])
export const modalPropsSelector = path(['app', 'modal', 'modalProps'])
export const notificationsSelector = path(['app', 'notifications'])
