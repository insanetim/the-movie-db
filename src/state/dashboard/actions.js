import * as types from './types'

export const fetchTrending = page => ({
  type: types.FETCH_TRENDING,
  payload: page
})

export const fetchTrendingRequest = () => ({
  type: types.FETCH_TRENDING_REQUEST
})

export const fetchTrendingSuccess = movies => ({
  type: types.FETCH_TRENDING_SUCCESS,
  payload: movies
})

export const fetchTrendingError = error => ({
  type: types.FETCH_TRENDING_ERROR,
  payload: error
})

export const setTrendingPage = page => ({
  type: types.SET_TRENDING_PAGE,
  payload: page
})

export const fetchSearch = ({ page, query }) => ({
  type: types.FETCH_SEARCH,
  payload: { page, query }
})

export const fetchSearchRequest = () => ({
  type: types.FETCH_SEARCH_REQUEST
})

export const fetchSearchSuccess = movies => ({
  type: types.FETCH_SEARCH_SUCCESS,
  payload: movies
})

export const fetchSearchError = error => ({
  type: types.FETCH_SEARCH_ERROR,
  payload: error
})

export const setSearchPage = page => ({
  type: types.SET_SEARCH_PAGE,
  payload: page
})
