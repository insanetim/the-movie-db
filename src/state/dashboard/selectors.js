import { path } from 'ramda'

export const trendingSelector = path(['dashboard', 'trending'])
export const searchSelector = path(['dashboard', 'search'])
