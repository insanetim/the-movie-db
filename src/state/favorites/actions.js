import * as types from './types'

export const fetchFavorites = page => ({
  type: types.FETCH_FAVORITES,
  payload: page
})

export const fetchFavoritesRequest = page => ({
  type: types.FETCH_FAVORITES_REQUEST,
  payload: page
})

export const fetchFavoritesSuccess = movies => ({
  type: types.FETCH_FAVORITES_SUCCESS,
  payload: movies
})

export const fetchFavoritesFailure = error => ({
  type: types.FETCH_FAVORITES_FAILURE,
  payload: error
})

export const setFavoritesPage = page => ({
  type: types.SET_FAVORITES_PAGE,
  payload: page
})
