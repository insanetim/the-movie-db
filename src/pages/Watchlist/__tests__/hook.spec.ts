import { act, renderHook } from '@testing-library/react'
import { Modal } from 'antd'
import { useSearchParams } from 'react-router-dom'
import mockAccount from 'src/__mocks__/mockAccount'
import { dispatch } from 'src/__mocks__/react-redux'
import { changeMovieInWatchlist } from 'src/store/movie/actions'
import { accountSelector } from 'src/store/session/selectors'
import { fetchWatchlist } from 'src/store/watchlist/actions'

import useContainer from '../hook'

jest.mock('src/store/watchlist/actions')

jest.mock('src/store/movie/actions')

jest.mock('src/store/session/selectors', () => ({
  accountSelector: jest.fn(() => null)
}))

jest.mock('src/store/watchlist/selectors', () => ({
  watchlistMoviesSelector: jest.fn(() => null)
}))

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useSearchParams: jest.fn()
}))
const searchParams = new URLSearchParams()
const setSearchParams = jest.fn()
jest.mocked(useSearchParams).mockReturnValue([searchParams, setSearchParams])

describe('Watchlist useContainer hook', () => {
  const confirmSpy = jest.spyOn(Modal, 'confirm')

  it('matches snapshot', () => {
    const { result } = renderHook(useContainer)

    expect(result.current).toMatchSnapshot()
  })

  it('checks `handlePagination` method', () => {
    const { result } = renderHook(useContainer)

    act(() => {
      result.current.handlePagination(3)
    })

    expect(setSearchParams).toHaveBeenCalledWith(
      new URLSearchParams({
        page: '3'
      })
    )
  })

  it('checks `handleMovieDelete` method', () => {
    let onOk = () => {
      return
    }
    const { result } = renderHook(useContainer)

    act(() => {
      onOk = result.current.handleMovieDelete(123, {
        stopPropagation: jest.fn()
      } as never)
    })
    onOk()

    expect(confirmSpy).toHaveBeenCalledWith({
      onOk,
      title: 'Do you want to delete movie from watchlist?'
    })
    expect(dispatch).toHaveBeenCalledWith(
      changeMovieInWatchlist({
        inWatchlist: false,
        movieId: 123
      })
    )
  })

  it('checks `useEffect` method with account', () => {
    jest.mocked(accountSelector).mockReturnValueOnce(mockAccount)
    renderHook(useContainer)

    expect(dispatch).toHaveBeenCalledWith(fetchWatchlist('1'))
  })
})
