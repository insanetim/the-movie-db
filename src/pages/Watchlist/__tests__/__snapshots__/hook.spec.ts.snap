// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Watchlist useContainer hook matches snapshot 1`] = `
{
  "error": null,
  "handleMovieDelete": [Function],
  "handlePagination": [Function],
  "loading": true,
  "movies": null,
}
`;
